app/controllers/puppetca_controller.rb

```diff
@@ -1,15 +1,13 @@
 class PuppetcaController < ApplicationController
+  rescue_from ProxyAPI::ProxyException do |exception|
+    process_error(:redirect => :back, :error_msg => exception.message)
```

Comment:
With the standard 500 page we get logging too, we should ensure the details of the exception are now logged too in this handler.